api/apps/geoprocessing/src/modules/planning-units/planning-units.service.ts

```diff
@@ -40,47 +41,88 @@ export class PlanningUnitsService {
   private readonly logger: Logger = new Logger(PlanningUnitsService.name);
   constructor(
     @InjectRepository(PlanningUnitsGeom)
-    private readonly protectedAreasRepository: Repository<PlanningUnitsGeom>,
+    private readonly planningUnitsRepository: Repository<PlanningUnitsGeom>,
     @Inject(TileService)
     private readonly tileService: TileService,
   ) {}
 
   /**
-   * @param bbox boundingbox of the area where the grids would be generated
    * @param planningUnitGridShape the grid shape that would be use for generating the grid. This grid shape
    * can be square or hexagon. If any grid shape is provided, square would be the default.
    * @param planningUnitAreakm2 area in km2 of the individual grid that would be generated.
    * If any value is not provided, 4000 would be the default.
    */
+  regularFuncionGridSelector(
+    planningUnitGridShape: PlanningUnitGridShape
+  ): string {
+    const functEquivalence: { [key in keyof typeof PlanningUnitGridShape]: string} = {
```

Comment:
alternatively `Record<keyof typeof PlanningUnitGridShape, string>` could be the type,
but actually, why it needs to match enum's keys instead of values? Was it intended?

Reply:
yes was intended. so we make sure this fails if new types of planning units are added to the enum and this is not changed